### DISCUSSION TITLE
Keep open folders state after tabnew

### DISCUSSION THREAD

**Discussion by BitInByte** [nvim-neo-tree/neo-tree.nvim#1129] [Q&A]
Hello everyone, first I want to start with a thank you with the amazing job with the Neo-tree.

I just have a question. I found out, that every time that I issue the `:tabnew` command, and then I open the neo-tree float, it always shows the directories close, even if I have had some directories open on some other tabs. Is there a way to keep this state from tab to tab so that it always keep the directories open from the last tab?

Thanks in advance for the help

**Comment by cseickel** [comment]
Unfortunately, keeping the state isolated between tabs was a core, very intentional design decision. There's no simple way to change that.

If you want to try to hack something together, you could try to copy the `state.explicitly_opened_directories` table from the state of the prior tab and set it on the state of the new tab before it renders. You can get the state objects with this method:

```lua
local manager = require("neo-tree.sources.manager")
local prev_state = manager.get_state("filesystem", <prev_tabid>)
local next_state = manager.get_state("filesystem", <next_tabid>)
new_state.explicitly_opened_directories = prev_state.explicitly_opened_directories
```
